DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt

```diff
@@ -0,0 +1,73 @@
+package dead.crumbs.ui
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.util.Log
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.IBinder
+
+class DeadReckoningService : Service(), SensorEventListener{
+
+    private lateinit var sensorManager: SensorManager
+    private var stepCounterSensor: Sensor? = null
+
+    override fun onCreate() {
+        super.onCreate()
+
+        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
+        stepCounterSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)
+
+        sensorManager.registerListener(this, stepCounterSensor, SensorManager.SENSOR_DELAY_NORMAL)
+
+    }
+
+    //----------Binding--------------
+    // Binder given to clients
+    private val binder = LocalBinder()
+
+    /**
+     * Class used for the client Binder.  Because we know this service always
+     * runs in the same process as its clients, we don't need to deal with IPC.
+     */
+
+    inner class LocalBinder : Binder(){
+        // Return this instance of LocalService so clients can call public methods
+        fun getService(): DeadReckoningService = this@DeadReckoningService
+    }
+
+    override fun onBind(intent: Intent?): IBinder? {
+        return binder
+    }
+
+    private var stepCounterInitial : Int = 0
+    private var stepCounter : Int = 0
+
+    /* This is called whenever this class (SensorEventListener) detects a new sensor value
+     * from a sensor it is listening to (registerListener) */
+    override fun onSensorChanged(event: SensorEvent) {
+        if (event.sensor.type == Sensor.TYPE_STEP_COUNTER){
+            val sensorValue = event.values[0]
+            // Set initial count value upon first reading
+            if (stepCounterInitial < 1)
+                stepCounterInitial = sensorValue.toInt()
+            // Update counter with #steps taken since initial value
+            stepCounter = sensorValue.toInt() - stepCounterInitial
+
+            //TODO: Find out how and where to pass value on to
```

Comment:
In my opinion I think that you should include values passing to a DeadReckoningViewModel. This would also allow deletion of this todo. But it has not been discussed or written before in regards to the associated trello task, so it is fair game if you choose not to.

If you choose to do then you can look for inspiration in branch "feature-DR-orientation"

Reply:
I think we'll find out what to do when we need to merge all the DR stuff